app/src/main/java/org/fossasia/openevent/app/data/RateLimiter.java

```diff
@@ -0,0 +1,40 @@
+package org.fossasia.openevent.app.data;
+
+import android.os.SystemClock;
+import android.support.v4.util.ArrayMap;
+
+import java.util.concurrent.TimeUnit;
```

Comment:
Will crash on lower APIs

Reply:
Will Duration work ? https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html
We are using Java8 so this shouldn't cause problems. Am I right ?

Reply:
ART != JDK

Android supports Java 8 Time API from API 24+

Reply:
Using ThreeTen back port for Duration should work. http://www.threeten.org/threetenbp/apidocs/org/threeten/bp/Duration.html

Reply:
Yes, you can use TimeUnit too if it is backported